src/Stratis.Bitcoin.Features.Miner/Staking/PosMinting.cs

```diff
@@ -1143,35 +1132,35 @@ public Models.GetStakingInfoModel GetGetStakingInfoModel()
-        /// but aiming not to exceed a total amount of coins bigger than (500+1)*3. S
+        /// Checks whether the coinstake should be split or not.
         /// </summary>
         /// <param name="stakedUtxosCount">Number of non-empty UTXOs currently at stake.</param>
         /// <param name="amountStaked">Total amount currently at stake</param>
         /// <param name="coinValue">Value of the coin we are considering to split</param>
         /// <param name="splitFactor">number of time we want to split the coin</param>
         /// <param name="chainHeight">Current height of the chain</param>
         /// <returns><c>true</c> if the coinstake should be split, <c>false</c> otherwise.</returns>
-        /// <remarks>The coinstake is split if the number of non-empty UTXOs we have in the wallet is under the given threshold, or if amount is above a certain value.</remarks>
-        /// <seealso cref="CoinstakeSplitLimitMultiplier"/>
+        /// <remarks>
+        /// The rule is that we want to be splitting coins until they are individually around 100 Strats,
+        /// but aiming not to exceed a total amount of coins bigger than (<c><seealso cref="requiredCoinAgeForStaking"/></c>>+1)*<c><seealso cref="CoinstakeSplitLimitMultiplier"/></c>.
+        /// </remarks>
         internal bool ShouldSplitStake(int stakedUtxosCount, long amountStaked, long coinValue, int splitFactor, int chainHeight)
         {
-            this.logger.LogTrace("({0}:{1},{2}:{3},{4}:{5},{6}:{7})", nameof(stakedUtxosCount), stakedUtxosCount, nameof(amountStaked), amountStaked, 
-                nameof(coinValue), coinValue, nameof(splitFactor), splitFactor);
+            this.logger.LogTrace("({0}:{1},{2}:{3},{4}:{5},{6}:{7},{8}:{9})", nameof(stakedUtxosCount), stakedUtxosCount, nameof(amountStaked), amountStaked, 
+                nameof(coinValue), coinValue, nameof(splitFactor), splitFactor, nameof(chainHeight), chainHeight);
             
             long coinAgeLimit = ((PosConsensusOptions)this.network.Consensus.Options).GetStakeMinConfirmations(chainHeight + 1, this.network);
-            long requiredCoinAgeForStaking = Math.Max(this.network.Consensus.CoinbaseMaturity, coinAgeLimit);
+            long coinMaturityLimit = this.network.Consensus.CoinbaseMaturity;
+            long requiredCoinAgeForStaking = Math.Max(coinMaturityLimit, coinAgeLimit);
             this.logger.LogTrace("Required coin age for staking is {0}.", requiredCoinAgeForStaking);
 
             long targetCoinDistributionSize = (requiredCoinAgeForStaking + 1) * CoinstakeSplitLimitMultiplier;
 
-            bool coinAboveMaxValue = coinValue > splitFactor * 100 * Money.COIN;
+            bool coinAboveMaxValue = coinValue > splitFactor * SplitCoinMinSize;
             bool utxosCanStillBeSplitted = stakedUtxosCount < targetCoinDistributionSize;
             bool coinAboveTargetAverage = coinValue > amountStaked / targetCoinDistributionSize;
```

Comment:
because of rounding i'd go with

`coinValue > (amountStaked / targetCoinDistributionSize) + Money.COIN`